project/Build.scala

```diff
@@ -433,6 +433,40 @@ object Build {
 
   private lazy val currentYear: String = java.util.Calendar.getInstance().get(java.util.Calendar.YEAR).toString
 
+  // Common generator for properties files
+  lazy val generatePropertiesFile(contents: String, file: File) = Def.task {
+    if (!(file.exists && IO.read(file) == contents)) {
+      IO.write(file, contents)
+    }
+    Seq(file)
+  }
+
+  // Generate compiler.properties consumed by sbt
+  lazy val generateCompilerProperties: Def.Initialize[Task[Seq[File]]] = {
+    import java.util._
+    import java.text._
+    val file = (Compile / resourceManaged).value / "compiler.properties"
+    val dateFormat = new SimpleDateFormat("yyyyMMdd-HHmmss")
+    dateFormat.setTimeZone(TimeZone.getTimeZone("GMT"))
+    val contents =                //2.11.11.v20170413-090219-8a413ba7cc
+      s"""version.number=${version.value}
+          |maven.version.number=${version.value}
+          |git.hash=${VersionUtil.gitHash}
+          |copyright.string=Copyright 2002-$currentYear, LAMP/EPFL
+        """.stripMargin
+    generatePropertiesFile(contents, file)
+  }
+
+  // Generate library3.properties consumed by scala.util.Properties
+  lazy val generateLibrary3Properties: Def.Initialize[Task[Seq[File]]] = generatePropertiesFile(
+    contents = s"""version.number=${version.value}
+                 |maven.version.number=${version.value}
+                 |copyright.string=Copyright 2002-$currentYear, LAMP/EPFL and Lightbend, Inc. dba Akka
+                 |shell.banner=%n     ________ ___   / /  ___%n    / __/ __// _ | / /  / _ |%n  __\\ \\/ /__/ __ |/ /__/ __ |%n /____/\\___/_/ |_/____/_/ | |%n                          |/  %s
```

Comment:
potentatially we could move it variable and interpolate so that it's readable for us